spec/services/resident_list_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+require_relative '../../app/services/resident_list'
+
+include AuthenticationSpecHelpers
+
+RSpec.describe ResidentList::PDF do
+  let(:pdf_file) { Rack::Test::UploadedFile.new('spec/fixtures/sample.pdf', 'application/pdf') }
+  let(:bad_pdf) { Rack::Test::UploadedFile.new('spec/fixtures/bad_sample.pdf', 'application/pdf')}
+  let(:duplicate_name_sample) { Rack::Test::UploadedFile.new('spec/fixtures/duplicate_name_sample.pdf', 'application/pdf')}
+
+  context 'when the PDF file is valid' do
+    it 'processes the uploaded PDF file' do
+      expect(ResidentList::PDF.new(pdf_file).load_residents.count).to eq(23)
+    end
+
+    it 'filters out unnecessary lines' do
+      residents = ResidentList::PDF.new(pdf_file).load_residents
+      expect(residents).not_to include(*ResidentList::PDF::SKIP_WORDS)
+    end
+
+    it 'skips over duplicate names' do
+      residents = ResidentList::PDF.new(duplicate_name_sample).load_residents
+      expect(residents.count).to eq(3)
+    end
+
+  end
+
+  context 'when the file is not a pdf' do
```

Comment:
⛏️ These three tests have some very long lines, it helps readability if you split them so there's a maximum of like 80 characters. Usually you can put the two arguments to `raise_error` on a new line, or you can make your block a multiline block with a `do end`, whichever you think is best for the situation.